src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeWriter.cs

```diff
@@ -651,5 +656,46 @@ public void Dispose()
             {
             }
         }
+
+        public struct StringLiteralWriter : IDisposable
```

Comment:
Totally get it. Only thing is you only ever truly use it once in `RuntimeBasicWriter`. Might as well be a method in there that takes a list of the tokens